compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2506,6 +2506,13 @@ static LogicalResult setRootConfig(mlir::FunctionOpInterface entryPointFn,
   SmallVector<int64_t> distTileSizes =
       getDefaultDistributedLevelTileSizes(op, DistributionHeuristicConfig{});
   TileSizesListType tileSizes = {distTileSizes};
+
+  // Add an extra level of tiling.
+  if (auto linalgOp = dyn_cast<linalg::LinalgOp>(*op)) {
+    SmallVector<int64_t> vecTileSizes = distTileSizes;
+    limitVectorTileSizes(linalgOp, vecTileSizes);
+    tileSizes.push_back(vecTileSizes);
+  }
```

Comment:
I think we'll need the second level of tiling on any TilingInterface ops, so please add a TODO. It is the key to get smaller problem size; you don't need the special logics in `Passes.cpp`. What I'd do is something like below

```suggestion
  SmallVector<int64_t> vecTileSizes = distTileSizes;
  // TODO: ...
  if (auto linalgOp = dyn_cast<linalg::LinalgOp>(*op)) {
    limitVectorTileSizes(linalgOp, vecTileSizes);
  }
  tileSizes.push_back(vecTileSizes);
```